src/Features/Core/Portable/ReplaceMethodWithProperty/ReplaceMethodWithPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,417 @@
+        {
+            var document = context.Document;
+            var service = document.GetLanguageService<IReplaceMethodWithPropertyService>();
+            if (service == null)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var position = context.Span.Start;
+            var token = root.FindToken(position);
+
+            if (!token.Span.Contains(context.Span))
+            {
+                return;
+            }
+
+            var methodDeclaration = service.GetMethodDeclaration(token);
+            if (methodDeclaration == null)
+            {
+                return;
+            }
+
+            // Ok, we're in the signature of the method.  Now see if the method is viable to be 
+            // replaced with a property.
+            var methodName = service.GetMethodName(methodDeclaration);
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodSymbol = semanticModel.GetDeclaredSymbol(methodDeclaration) as IMethodSymbol;
+            if (!IsValidGetMethod(methodSymbol))
+            {
+                return;
+            }
+
+            var hasGetPrefix = HasGetPrefix(methodName);
+            var propertyName = hasGetPrefix
+                ? methodName.Substring(GetPrefix.Length)
+                : methodName;
+            var nameChanged = hasGetPrefix;
+
+            // Looks good!
+            context.RegisterRefactoring(new ReplaceMethodWithPropertyCodeAction(
+                string.Format(FeaturesResources.Replace0WithProperty, methodName),
+                c => ReplaceMethodsWithProperty(context.Document, propertyName, nameChanged, methodSymbol, setMethod: null, cancellationToken: c),
+                methodName));
+
+            // If this method starts with 'Get' see if there's an associated 'Set' method we could 
+            // replace as well.
+            if (hasGetPrefix)
+            {
+                var setMethod = FindSetMethod(methodSymbol);
+                if (setMethod != null)
+                {
+                    context.RegisterRefactoring(new ReplaceMethodWithPropertyCodeAction(
+                        string.Format(FeaturesResources.Replace0and1WithProperty, methodName, setMethod.Name),
+                        c => ReplaceMethodsWithProperty(context.Document, propertyName, nameChanged, methodSymbol, setMethod, cancellationToken: c),
+                        methodName + "-get/set"));
+                }
+            }
+        }
+
+        private static bool HasGetPrefix(SyntaxToken identifier)
+        {
+            return HasGetPrefix(identifier.ValueText);
+        }
+
+        private static bool HasGetPrefix(string text)
+        {
+            return text.StartsWith(GetPrefix) && text.Length > GetPrefix.Length;
```

Comment:
This should check for the next character being upper case or an underscore.  If I have a method `int GetterCount()` I don't want a property `int terCount { get; set; }`.